### PR TITLE
use case insensitive matching for http request header keys

### LINKED ISSUE #1226: Case sensitive comparison of some headers causes issues in HTTP/2 environments
When moving from Orion-LD 1.0.1 to 1.1.0 we experienced a regression where our solution worked perfectly in a docker composed environment, but failed mysteriously on Kubernetes using an istio service mesh. The problem has been tracked down to the fact that the Accept header is no longer respected, when invoking GET requests for /ngsi-ld/v1/entities or /ngsi-ld/v1/entities/{entityid}, in environments where headers are automatically lowercased (i.e. when using HTTP/2).

The behaviour can easily be reproduced by using curl and passing -H "accept: application/ld+json" to one of the above endpoints. The broker will then respond with Content-Type: application/json instead. A lower case -H "link: ...." will be respected though...

The new behaviour seems to have been introduced in https://github.com/FIWARE/context.Orion-LD/pull/1001 where the accept header handling was rewritten. A lot of headers are properly compared using strcasecmp, but a few are compared using strcmp which causes the code to fail when Accept is being received as accept.

See for example https://github.com/FIWARE/context.Orion-LD/blob/80d6c410700ee32416ae916de9ccc97f91f47877/src/lib/orionld/rest/orionldMhdConnectionInit.cpp#L416

### PR DESCRIPTION
**Pull request by istyf**
## Proposed changes

This PR fixes https://github.com/FIWARE/context.Orion-LD/issues/1226 by replacing strcmp with strcasecmp for the four request headers Orionld-Legacy, Performance, NGSILD-Scope and Accept.

## Types of changes

What types of changes does your code introduce to the project: _Put an `x` in the boxes that apply_

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before
merging your code._

-   [x] I have read the [CONTRIBUTING](https://github.com/FIWARE/context.Orion-LD/blob/master/CONTRIBUTING.md) doc
-   [x] I have signed the [CLA](https://fiware.github.io/contribution-requirements/individual-cla.pdf)
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)
-   [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

I should have added a test for this, but I don't have the correct toolchain setup currently, and wanted to get the issue and PR posted as soon as possible in case there are others that get affected by this.

I have read the CLA Document and I hereby sign the CLA.

### CODE REVIEW

**Comment by istyf** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by kzangeli — APPROVED** [review]
Well spotted, thanks!

**Comment by kzangeli** [comment]
Seems like the tests are halted due to something I need to do ...
I really don't know what I need to do.  I've already accepted the PR.

If no progress on this before, say Tuesday, I'll merge it anyway.
The fix is quite straightforward and I have no doubt whatsoever.   It's GOOD.

Would be nice to see the test suite run though ...

**Comment by kzangeli** [comment]
So, our DevOps expert tells me you need to "enable the Actions on the fork", for the tests to run.
Could you please try that, and we'll see what happens ...

**Comment by istyf** [comment]
Something along these lines? https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

It seems to me that a repo maintainer needs to that, and not the person doing the fork. Right?

**Comment by kzangeli** [comment]
Only, there isn't any "Workflow(s) awaiting approval". Really don't know what to do.

According to this [SOF issue](https://stackoverflow.com/questions/52200096/github-pull-request-waiting-for-status-to-be-reported) it seems like it's a bug in github ...

**Pull request merged**
